support-services/src/main/scala/com/gu/support/redemption/DynamoTableAsync.scala

```diff
@@ -0,0 +1,86 @@
+package com.gu.support.redemption
+
+import java.util.concurrent.CompletionException
+
+import com.typesafe.scalalogging.LazyLogging
+import software.amazon.awssdk.auth.credentials.{AwsCredentialsProviderChain, InstanceProfileCredentialsProvider, ProfileCredentialsProvider}
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, GetItemRequest, UpdateItemRequest}
+
+import scala.collection.JavaConverters._
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+object DynamoTableAsync {
```

Comment:
Maybe call this `RedemptionTableAsync` and the classes below `RedemptionLookup` or `RedemptionDynamoLookup` etc.? 

I'm not entirely sure but Dynamo feels a bit too generic. We already have a `DynamoTables` class in the promotions namespace for instance.

Reply:
there isn't anything redemption specific in the code (and intentionally so for testing and separation reasons), it's only exists to put a simplified and scala-friendly facade on dynamo async table api to make the business logic nicer.
Yes in reality the scala api for it is only used by redemptions, but that is not relevant to the class itself (and that's why it's in the `com.gu.support.redemption` package)
If the need happened, I'd be happy to pull it out of the package and use it in other places, perhaps the promo codes stuff will use this same code if we update it to aws-sdk v2.
Adding the word redemption encourages redemption specific code to end up here.
Makes sense or still prefer a rename?

Reply:
Yes that's fine